public/js/index.js

```diff
@@ -0,0 +1,207 @@
+$(document).ready(() => {
+  let uploadedFiles = [];
+
+  $('#table-display-error').hide();
+  $('#upload-error').hide();
+  $('.file-index').hide();
+  $('.file-upload').fadeIn('slow');
+
+  // upload file button
+  $('#button-upload-file').click(() => {
+    $('#button-upload-file').removeAttr('disabled');
+    $('#select-file').empty();
+
+    if (uploadedFiles.length < 5) {
+      $('#button-upload-file').removeAttr('disabled');
+      $('#upload-error').hide();
+
+      uploadedFiles = $('#image-file')[0].files;
+      updateSelectOptions(uploadedFiles, '#select-file');
+      updateSelectOptions(uploadedFiles, '#select-file-search');
+    } else if (uploadedFiles.length === 0) {
+      $('#button-upload-file').attr('disabled', true);
+      $('#upload-error').show();
+    } else {
+      $('#button-upload-file').attr('disabled', true);
+      $('#upload-error').show();
+    }
+
+    $('.file-upload').hide();
+    $('.file-index').fadeIn('slow');
+  });
+
+  // create index button
+  $('#button-create-index').click(() => {
+    const uploadedFileName = $('#select-file').val();
+    const uploadedFile = getSelectOptionFile(uploadedFileName, uploadedFiles);
+
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      const data = e.target.result;
+      let invertedIndex = new InvertedIndex(JSON.parse(data));
+
+      if (invertedIndex.checkIfJson(uploadedFileName)) {
+        if (invertedIndex.containsTitleText()) {
+          const contentToDisplay = invertedIndex.displayInTableFormat();
```

Comment:
I like the decomposition of the work here (and below) into smaller functions that you just call. Makes the code clear and readable. 👍